Updated and tested to complete mapping with Blocks environment.

okyspace opened the pull request:
In general works, less the ROS melodic installation. 
For ROS melodic installation, dependencies issue still exist and adding on a few will help and complete docker build. But the added on dependencies do not seems to be complete as subsequent clean build will face similar issue too.

okyspace committed:
Updated and tested. Able to complete mapping with Blocks environment.

Added cloning of TartanAir repo in docker, copied settings.json, reorganised related functions together, configurable github handle, AirSim1.4,  added some dependencies to fix some non-critical errors and some cosmetic edits to view each section easily. Pending fix for ROS melodic installation.

okyspace committed:
Added some minor point to note

jonyktan commented on README.md:
```diff
@@ -60,4 +60,7 @@
+1. Unreal Engine will prompt to rebuild AirSim. Click "Yes". First compile will take awhile.
+
+## Others
+1. Remember to edit the blocks.sh script path for saving map.
```
Which step of the Wenshan instructions does this correspond to?

okyspace replied:
This corresponds to the mapping process using `sh blocks.sh`. When mapping is done, it saves to a path indicated in the blocks.sh which is hardcoded  to /home/wenshan.... thus directory not found.

okyspace replied:
If we agree to shift tartanair workspace to mounted volume (see comments below), this is then too specific. Can omit from this repo, since this repo should encompass only building docker image, not every single details of running the various processes?

okyspace replied:
Removed. Will have tartanair workspace in docker vol.

jonyktan commented on Dockerfile:
```diff
@@ -70,142 +78,121 @@ RUN sudo apt install -y --no-install-recommends \
 
+# Copy settings_mapping.json to ~Document/AirSim/settings.json
+WORKDIR /home/$USERNAME/Documents/AirSim
+RUN sudo mv /$FOLDER_NAME/$WORKSPACE/src/sample_pipeline/settings_mapping.json /home/$USERNAME/Documents/AirSim/settings.json
```
Thanks! May want to leave this out of the dockerfile though, cause the data collection part requires a sample_pipeline/settings_data.json as AirSim/settings.json instead.

okyspace replied:
Agreed, will remove this.

okyspace replied:
Removed.

jonyktan commented on Dockerfile:
```diff
@@ -70,142 +78,121 @@ RUN sudo apt install -y --no-install-recommends \
+# ==============================
+# Unreal Engine 4.25
+# ==============================
+ARG GITHUB_USER=<update>
```
Thinking whether to put this as part of the README instructions instead. So docker build instruction also specifies these args, so user don't have to go into dockerfile to edit. Same for GITHUB_PWD

okyspace replied:
Agreed, I can change this too.

okyspace replied:
Done for README and Dockerfile.

jonyktan commented on Dockerfile:
```diff
@@ -70,142 +78,121 @@ RUN sudo apt install -y --no-install-recommends \
 RUN ./setup.sh
 RUN ./build.sh
 # use ./build.sh --debug to build in debug mode
+# TODO: Copy Plugins folder to host if possible
```
How come need to copy to host?

okyspace replied:
Not really needed, just thought to "auto" copy to host when we build this docker image, so we can just add this Plugin folder to our own UE envs. Will remove this TODO.

okyspace replied:
Removed.

jonyktan commented on Dockerfile:
```diff
@@ -70,142 +78,121 @@ RUN sudo apt install -y --no-install-recommends \
+RUN curl -sSL 'http://keyserver.ubuntu.com/pks/lookup?op=get&search=0xC1CF6E31E6BADE8868B172B4F42ED6FBAB17C654' | sudo apt-key add -
+WORKDIR /home/$USERNAME
+# Pre-requisites for ros-melodic-desktop-full
+RUN sudo apt-get update && sudo apt-get install ros-melodic-perception ros-melodic-simulators ros-melodic-urdf-sim-tutorial ros-melodic-perception-pcl ros-melodic-pcl-conversions ros-melodic-pcl-ros libgazebo9-dev ros-melodic-gazebo-dev ros-melodic-gazebo-plugins ros-melodic-gazebo-ros ros-melodic-gazebo-ros-pkgs -y --no-install-recommends
```
Did this instruction work for you? I put this in because last week I needed to do this. But recently Melvin and I both built, this instruction didn't work for us and we had to revert to RUN sudo apt-get install ros-melodic-desktop-full

okyspace replied:
Either didn't work for me when I build from clean state, ie. remove all docker images, containers. What I tried was when I hit dependencies issue, I try to add one dependency at a time and try again. Seems like it works after adding some (not all) dependencies. 

Need to troubleshoot further though.

okyspace replied:
Tested again and it works.

jonyktan replied:
I tested again last night too. Worked with apt-get install ros-melodic-desktop-full haha... I think can leave both in, write comment for user to choose whichever works for their host.

okyspace replied:
Done!

jonyktan reviewed (commented):
Thanks a lot! Had some comments for discussion. Will rebuild using this dockerfile soon!

okyspace commented:
Actually thinking if we should move tartanair workspace out of docker image, since we supposed to touch (improve / add) the codes. Thinking to add into the vol that we mount to the docker container. If this is ok, we can rename the folders in docker image to something more relevant for both UE env and tartanair workspace.

okyspace committed:
Removed copy settings.json.

okyspace committed:
Remove TartanAir to be mounted from vol instead. Updated Github handles to be passed in using --build-arg.

okyspace committed:
Updated to incorporate passing in Github handle as arg.

okyspace committed:
Added networkx, removed folders. 

Folders removed as preferred to store in persisted docker vol.

okyspace commented:
Also removed folders for map, etc since I think we want to persist these when container is killed. suggest shift to docker vol.

jonyktan commented on Dockerfile:
```diff
@@ -1,9 +1,10 @@
 ### Instructions for Docker
+## set your github user and password in this dockerfile. search for <update>
```
Can delete this line please?

okyspace replied:
Removed

jonyktan commented on Dockerfile:
```diff
@@ -13,25 +14,41 @@
+# ==============================
+# Configurable params
+# ==============================
+ARG GITHUB_USER
```
I think can delete this if declared in docker build? Does leaving it in without declaring, cause errors instead?

okyspace replied:
This is needed if not dockerfile won't register the passed-in arg using --build-arg.

jonyktan commented on Dockerfile:
```diff
@@ -13,25 +14,41 @@
+# ==============================
+ARG GITHUB_USER
+ARG GITHUB_PWD
+RUN env
```
What's this for?

okyspace replied:
This is needed if not dockerfile won't register the passed-in arg using --build-arg. Can remove RUN env though, it is mainly to confirm the ARG are registered.

okyspace committed:
Removed unnecessary comment.

okyspace committed:
Added comment for user to decide which option to use for ROS installation. 

Option 1: ros-melodic-full.
Option 2: With pre-requisites installed before ros-melodic-full.

jonyktan committed:
Merge branch 'main' into add-tartanair-repo
